examples/sparse/hypergraphatt.py

```diff
@@ -0,0 +1,126 @@
+"""
+Hypergraph Convolution and Hypergraph Attention
+(https://arxiv.org/pdf/1901.08150.pdf).
+"""
+import dgl
+import dgl.data
+import dgl.mock_sparse as dglsp
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchmetrics.functional import accuracy
+import tqdm
+
+# Question:
+# The paper proposed two models, but HypergraphConv is essentially
+# what is in the HGNN paper (identical).  So should I rewrite HGNN
+# in the same fashion as this file?
+def hypergraph_laplacian(H):
+    ###########################################################
+    # (HIGHLIGHT) Compute the Laplacian with Sparse Matrix API
+    ###########################################################
+    d_V = H.sum(1)  # node degree
+    d_E = H.sum(0)  # edge degree
+    n_edges = d_E.shape[0]
+    D_V_invsqrt = dglsp.diag(d_V ** -0.5)  # D_V ** (-1/2)
+    D_E_inv = dglsp.diag(d_E ** -1)  # D_E ** (-1)
+    W = dglsp.identity((n_edges, n_edges))
+    laplacian = D_V_invsqrt @ H @ W @ D_E_inv @ H.T @ D_V_invsqrt
+
+    return laplacian
+
+class HypergraphAttention(nn.Module):
+    """Hypergraph Attention module.
+
+    As per Section 3.3 in the paper
+    `Hypergraph Convolution and Hypergraph Attention
+    <https://arxiv.org/pdf/1901.08150.pdf>`_, this module only works when
+    the number of hyperedges equal the number of nodes.
```

Comment:
```suggestion
    the number of hyperedges equals to the number of nodes.
```

Reply:
This description is actually outdated as I included the hyperedge representations as input as well (same as PyG).  So I'll remove it.